SocialCareCaseViewerApi/V1/Controllers/SocialCareCaseViewerApiController.cs

```diff
@@ -102,6 +104,50 @@ public IActionResult AddNewResident([FromBody] AddNewResidentRequest residentReq
             }
         }
 
+        /// <summary>
+        /// Get resident by id
+        /// </summary>
+        /// <response code="200">Success</response>
+        /// <response code="400">One or more request parameters are invalid or missing</response>
+        /// <response code="500">There was a problem getting the record</response>
+        ///
+        [ProducesResponseType(typeof(GetPersonResponse), StatusCodes.Status200OK)]
+        [HttpGet]
+        [Route("residents/{id}")]
+        public IActionResult GetPerson([FromQuery] GetPersonRequest request)
+        {
+            var response = _personUseCase.ExecuteGet(request);
+
+            if (response == null)
+            {
+                return NotFound();
+            }
+
+            return StatusCode(200, response);
+        }
+
+        /// <summary>
+        /// Update resident details
+        /// </summary>
+        /// <response code="200">Success</response>
+        /// <response code="400">One or more request parameters are invalid or missing</response>
+        /// <response code="500">There was a problem updating the records</response>
+        ///
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [HttpPatch]
+        [Route("residents")]
+        public IActionResult UpdatePerson([FromBody] UpdatePersonRequest request)
+        {
+            var response = _personUseCase.ExecutePatch(request);
+
+            if (response?.Message == "PersonNotFound")
```

Comment:
Instead of this response being returned do we want to just throw an UpdatePersonException in the gateway if no person is found, then utilise try catch and return NotFound with the exception message.

This way UseCase doesn't have to return anything and we're aren't validating based on strings being specific values.

Reply:
Implemented the exception approach to tidy this up.